framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -277,9 +279,20 @@ protected TreeAnnotator createTreeAnnotator() {
      * @return a type annotator
      */
     protected TypeAnnotator createTypeAnnotator() {
-        implicitsTypeAnnotator = new ImplicitsTypeAnnotator(this);
+        List<TypeAnnotator> typeAnnotators = new ArrayList<>();
+        RelevantJavaTypes relevantJavaTypes =
+                checker.getClass().getAnnotation(RelevantJavaTypes.class);
+        if (relevantJavaTypes != null) {
+            Class<?>[] classes = relevantJavaTypes.value();
+            typeAnnotators.add(
+                    new IrrelevantTypeAnnotator(
+                            this, getQualifierHierarchy().getTopAnnotations(), classes));
+        }
 
-        return new ListTypeAnnotator(new PropagationTypeAnnotator(this), implicitsTypeAnnotator);
+        implicitsTypeAnnotator = new ImplicitsTypeAnnotator(this);
+        typeAnnotators.add(implicitsTypeAnnotator);
+        typeAnnotators.add(new PropagationTypeAnnotator(this));
```

Comment:
Isn't the PropagationTypeAnnotator at the end now, instead of previously at the beginning?
Can you add some comments what the intuition behind the ordering is?